circuit_knitting/cutting/find_cuts.py

```diff
@@ -0,0 +1,143 @@
+# This code is a Qiskit project.
+
+# (C) Copyright IBM 2024.
+
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Function for automatically finding locations for gate and wire cuts."""
+
+from __future__ import annotations
+
+from typing import cast, Any
+
+from qiskit.circuit import QuantumCircuit, CircuitInstruction
+
+from .instructions import CutWire
+from .cutting_decomposition import cut_gates
+from .cut_finding.optimization_settings import (
+    OptimizationSettings,
+    OptimizationParameters,
+)
+from .cut_finding.quantum_device_constraints import DeviceConstraints
+from .cut_finding.disjoint_subcircuits_state import DisjointSubcircuitsState
+from .cut_finding.circuit_interface import SimpleGateList
+from .cut_finding.lo_cuts_optimizer import LOCutsOptimizer
+from .cut_finding.cco_utils import qc_to_cco_circuit
+
+
+def find_cuts(
+    circuit: QuantumCircuit,
+    optimization: OptimizationParameters,
+    constraints: DeviceConstraints,
+) -> tuple[QuantumCircuit, dict[str, float]]:
+    """Find cut locations in a circuit, given optimization settings and QPU constraints.
+
+    Args:
+        circuit: The circuit to cut. The circuit must contain only single two-qubit
+            gates.
+        optimization: Options for controlling optimizer behavior. Currently, the optimal
+            cuts are arrived at using Dijkstra's best-first search algorithm. The specified
+            parameters are:
+
+            - max_gamma: Specifies a constraint on the maximum value of gamma that a
```

Comment:
We probably don't need to explicitly state all this in the docstring anymore since we are exposing the settings classes in the docs